### PR TITLE
Add an evaluator for Estrin's method; move the evaluators to a separate file and test them separately

### PR DESCRIPTION
**Pull request by pleroy**

### CODE REVIEW

File: numerics/polynomial_evaluators_body.hpp [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+namespace {
+
+// Greatest power of 2 less than or equal to n.  8 -> 8, 7 -> 4.
+constexpr int flp2(int const n) {
+  return n == 0 ? 0 : n == 1 ? 1 : flp2(n >> 1) << 1;
+}
+
+// Ceiling log2 of n.  8 -> 3, 7 -> 2.
+constexpr int log2(int const n) {
```
Comment (eggrobin) [inline numerics/polynomial_evaluators_body.hpp]: Why not explicitly CeilingLog2? also why the styleguide-violating lowercase?

File: numerics/polynomial_evaluators_body.hpp [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+#include <tuple>
+
+namespace principia {
+namespace numerics {
+namespace internal_polynomial_evaluators {
+
+namespace {
+
+// Greatest power of 2 less than or equal to n.  8 -> 8, 7 -> 4.
+constexpr int flp2(int const n) {
```
Comment (eggrobin) [inline numerics/polynomial_evaluators_body.hpp]: GreatestPowerOf2LessThan(n)?

File: numerics/polynomial_evaluators_body.hpp [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+
+template<typename Argument, int... orders>
+auto SquaresGenerator<Argument, std::integer_sequence<int, orders...>>::
+    Evaluate(Argument const& argument) -> Type {
+  return std::make_tuple(
+      SquareGenerator<Argument, orders>::Evaluate(argument)...);
+}
+
+template<typename Value, typename Argument, int degree, int low, int high>
+struct InternalEstrinEvaluator {
```
Comment (eggrobin) [inline numerics/polynomial_evaluators_body.hpp]: This needs comment: what does the `Evaluate` function evaluate? what are `low` and `high`?

File: numerics/polynomial_evaluators_body.hpp (hunk 2 above)
Comment (pleroy) [reply numerics/polynomial_evaluators_body.hpp]: Used `FloorPowerOf2'.

File: numerics/polynomial_evaluators_body.hpp (hunk 1 above)
Comment (pleroy) [reply numerics/polynomial_evaluators_body.hpp]: Done.

File: numerics/polynomial_evaluators_body.hpp (hunk 3 above)
Comment (pleroy) [reply numerics/polynomial_evaluators_body.hpp]: Done.

File: numerics/polynomial_evaluators_body.hpp [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+#include <tuple>
+
+namespace principia {
+namespace numerics {
+namespace internal_polynomial_evaluators {
+
+namespace {
+
+// Greatest power of 2 less than or equal to n.  8 -> 8, 7 -> 4.
+constexpr int FloorPowerOf2(int const n) {
```
Comment (eggrobin) [inline numerics/polynomial_evaluators_body.hpp]: RoundToward0To1Bit

Comment (eggrobin) [reply numerics/polynomial_evaluators_body.hpp]: Or FloorToPowerOf2, but there needs to be a To somewhere

Comment (pleroy) [reply numerics/polynomial_evaluators_body.hpp]: Done.

**Pull request merged**